### PR TITLE
tests: fix fs tests run w/o full getdents support

### PR DESCRIPTION
**Pull request by whitlockjc**
fs_access, fs_async_dir and fs_symlink_dir could fail when the
filesystem where the tests are run does not fully support getting the
file type via getdents.  Having getdents support does not imply that the
underlying filesystem fully supports what is necessary for these tests
to have passed prior to this commit.  So the failing tests were updated
to check for UV_FILE as it did previously but it will also check for
UV_UNKNOWN as well to handle the filesystems that do not fully support
getting the file type via getdents.  For OS X and Windows, which are
known to fully support getting the file type, the tests work as they did
before and will not check for UV_UNKNOWN.  We could/should update the
preprocessor directive that chooses "vigorous" or "lax" checks
accordingly when we learn of new environments that should always do the
original "vigorous" checks.

Fixes: #501

### CODE REVIEW

**Comment by whitlockjc** [comment]
Don't create a CI for this yet.  Investigating something.

**Comment by whitlockjc** [comment]
This PR should be good for CI.  Tested on OS X and CentOS.

**Comment by saghul** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv+any-pr+multi/143/

**Comment by whitlockjc** [comment]
RHEL 7 seems good: https://ci.nodejs.org/view/libuv/job/libuv+any-pr+multi/143/nodes=centos7-64/console  I looked at FreeBSD, OS X, Ubuntu and Windows and their test failures were unrelated.  I look forward to the review process.

File: test/test-fs.c [hunk 1]
```diff
@@ -427,6 +427,28 @@ static void rmdir_cb(uv_fs_t* req) {
...
+     *     http://man7.org/linux/man-pages/man2/getdents.2.html
+     *     https://github.com/libuv/libuv/issues/501
+     */
+    #if defined(__APPLE__) || defined(_WIN32)
+      ASSERT(dent.type == UV_DIRENT_FILE);
+    #else
+      ASSERT(dent.type == UV_DIRENT_FILE || dent.type == UV_DIRENT_UNKNOWN);
+    #endif
+  #else
+      ASSERT(dent.type == UV_DIRENT_UNKNOWN);
```
Comment (saghul) [inline test/test-fs.c]: 2 space indent.

**Comment by saghul** [comment]
One femto-nit, but LGTM. Great work @whitlockjc :+1:

File: test/test-fs.c (hunk 1 above)
Comment (whitlockjc) [inline test/test-fs.c]: That was a relic of moving this from one of the places with an extra level of indentation.  ;)  I'll update the PR.

**Comment by whitlockjc** [comment]
Should be fixed.  Let me know if there is more to do.

**Comment by saghul** [comment]
Landed in 30fc362, thanks a lot! :heart_eyes:

**Comment by whitlockjc** [comment]
Any time.

**Pull request closed without merging**